From 8f9fcab47a1ac1457ac08d24b2e146f37082ea29 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 29 Nov 2023 23:50:05 +0100
Subject: [PATCH 70/N] fixup! Respect the `db_home: env` setting under more
 circumstances

In preparation for integrating the patches that made it upstream via
https://inbox.sourceware.org/cygwin-patches/cover.1684753872.git.johannes.schindelin@gmx.de/,
let's revert the original patches from Git for Windows' branch thicket.

This reverts commit f0d56412872d931e55fde9406d390e2070f2befb.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/uinfo.cc | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/winsup/cygwin/uinfo.cc b/winsup/cygwin/uinfo.cc
index 1dbd826..00a447d 100644
--- a/winsup/cygwin/uinfo.cc
+++ b/winsup/cygwin/uinfo.cc
@@ -998,8 +998,6 @@ cygheap_pwdgrp::get_home (PUSER_INFO_3 ui, cygpsid &sid, PCWSTR dom,
 
   for (uint16_t idx = 0; !home && idx < NSS_SCHEME_MAX; ++idx)
     {
-      if (!ui && home_scheme[idx].method != NSS_SCHEME_ENV)
-        continue;
       switch (home_scheme[idx].method)
 	{
 	case NSS_SCHEME_FALLBACK:
@@ -2138,9 +2136,6 @@ pwdgrp::fetch_account_from_windows (fetch_user_arg_t &arg, cyg_ldap *pldap)
 	{
 	  /* Just some fake. */
 	  sid = csid.create (99, 1, 0);
-	  if (arg.id == cygheap->user.real_uid)
-	    home = cygheap->pg.get_home(NULL, cygheap->user.sid(),
-	      NULL, NULL, false);
 	  break;
 	}
       else if (arg.id >= UNIX_POSIX_OFFSET)
@@ -2690,11 +2685,10 @@ pwdgrp::fetch_account_from_windows (fetch_user_arg_t &arg, cyg_ldap *pldap)
      logon.  Unless it's the SYSTEM account.  This conveniently allows to
      logon interactively as SYSTEM for debugging purposes. */
   else if (acc_type != SidTypeUser && sid != well_known_system_sid)
-    __small_sprintf (linebuf, "%W:*:%u:%u:U-%W\\%W,%s:%s:/sbin/nologin",
+    __small_sprintf (linebuf, "%W:*:%u:%u:U-%W\\%W,%s:/:/sbin/nologin",
 		     posix_name, uid, gid,
 		     dom, name,
-		     sid.string ((char *) sidstr),
-		     home ? home : "/");
+		     sid.string ((char *) sidstr));
   else
     __small_sprintf (linebuf, "%W:*:%u:%u:%s%sU-%W\\%W,%s:%s%W:%s",
 		     posix_name, uid, gid,
